support/azureutil/azureutil.go

```diff
@@ -217,3 +226,64 @@ func VerifyResourceGroupLocationsMatch(ctx context.Context, hc *hyperv1.HostedCl
 
 	return nil
 }
+
+// GetAzureCredentialsFromSecret gets the Service Principal client ID, client secret, and tenant ID from the credentials
+// secret. This function will be modified a bit once the Microsoft sidecar containers support Managed Identity are
+// delivered (expected Oct 2024).
+func GetAzureCredentialsFromSecret(ctx context.Context, c client.Client, namespace, credsName string) (*corev1.Secret, error) {
+	var azureCredentials corev1.Secret
+
+	// Retrieve the Azure credentials secret to extract the needed fields for the managed identity containers
+	credentialsSecretName := client.ObjectKey{Namespace: namespace, Name: credsName}
+	if err := c.Get(ctx, credentialsSecretName, &azureCredentials); err != nil {
+		return nil, fmt.Errorf("failed to get secret %s: %w", credentialsSecretName, err)
+	}
+
+	for _, expectedKey := range []string{"AZURE_CLIENT_ID", "AZURE_CLIENT_SECRET", "AZURE_TENANT_ID"} {
+		if _, found := azureCredentials.Data[expectedKey]; !found {
+			return nil, fmt.Errorf("credentials secret for cluster doesn't have required key %s", expectedKey)
+		}
+	}
+
+	return &azureCredentials, nil
+}
+
+// AdapterInitContainer returns the Microsoft adapter-init init container. This container needs the NET_ADMIN permission
+// so the adapter-server sidecar container can intercept the Managed Identity Azure API authentication calls.
+func AdapterInitContainer() corev1.Container {
+	return corev1.Container{
+		Name:            "adapter-init",
+		Image:           AdapterInitImage,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		SecurityContext: &corev1.SecurityContext{
+			Capabilities: &corev1.Capabilities{
+				Add: []corev1.Capability{
+					"NET_ADMIN",
+				},
+			},
+		}}
+}
+
+// AdapterServerContainer returns the Microsoft adapter-server sidecar container. Currently, this container mimics Azure
+// Managed Identity approval and returns an authentication token. The container currently needs a Service Principal to
+// do this. Future versions of this container will be able to take a Managed Identity instead.
+func AdapterServerContainer(clientID, clientSecret, tenantID string) corev1.Container {
```

Comment:
This func is not being used, is that for a future implementation?

Reply:
Yeah, this is listed in the PR description - Follow on work to integrate these functions and implement managed identity in HCP will be done in [OCPBUGS-42434](https://issues.redhat.com/browse/OCPBUGS-42434).